doc/internal/consensus/Consensus-Protocol.md

```diff
@@ -0,0 +1,127 @@
+# The consensus
+
+Before talking about consensus, we need to discuss some logistics based on how the systems can co-exist.
+
+* Communication: Distributed systems need a method to communicate between each other. Remote Procedure Calls is the mechanism using which a standalone server can talk to another. The existing `network` layer of the database will handle all the communication between servers. 
+* Security: Access control mechanisms need to be in place to decide on access to functions in the servers based on their state (leader, follower, candidate)
```

Comment:
Not a raft problem. We need to implement an access control layer that is configured with SQL statements, such as in common databases. They are communicated over raft (replicating statements), but not part of the consensus protocol however.

Reply:
What I meant were raft functions.

There are variables that only the leader can maintain.